test/mcmc/gibbs.jl

```diff
@@ -135,10 +281,271 @@ using Turing.RandomMeasures: ChineseRestaurantProcess, DirichletProcess
                 m[k] ~ Normal(1.0, 1.0)
             end
         end
-        model = imm(randn(100), 1.0)
+        model = imm(Random.randn(100), 1.0)
         # https://github.com/TuringLang/Turing.jl/issues/1725
-        # sample(model, Gibbs(MH(:z), HMC(0.01, 4, :m)), 100);
-        sample(model, Gibbs(PG(10, :z), HMC(0.01, 4, :m; adtype=adbackend)), 100)
+        # sample(model, Gibbs(; z=MH(), m=HMC(0.01, 4)), 100);
+        sample(model, Gibbs(; z=PG(10), m=HMC(0.01, 4; adtype=adbackend)), 100)
+    end
+
+    @testset "dynamic model with dot tilde" begin
+        @model function dynamic_model_with_dot_tilde(num_zs=10)
+            z = Vector(undef, num_zs)
+            z .~ Exponential(1.0)
+            num_ms = Int(round(sum(z)))
+            m = Vector(undef, num_ms)
+            return m .~ Normal(1.0, 1.0)
+        end
```

Comment:
Maybe add typing for this?
```suggestion
        @model function dynamic_model_with_dot_tilde(num_zs=10, ::Type{TV}=Vector{Float64}) where {TV}
            z = TV(undef, num_zs)
            z .~ Exponential(1.0)
            num_ms = Int(round(sum(z)))
            m = TV(undef, num_ms)
            return m .~ Normal(1.0, 1.0)
        end
```

Reply:
Also, is tehre a reason why you're doing `Exponential` and converting rather than just doing something like `Poisson`, i.e. discrete distribution?